### ISSUE TITLE
warning C4251 due to missing dll-interface in TagLib::FileName

### ISSUE THREAD

**Issue by daschuer** [taglib/taglib#1185]
With MSVC 14.29.30133 (Visual Studio 2019) I get the following warning. 

```
Warning: C:\buildenv\mixxx-deps-rel-2.4-x64-windows-3e909e2\installed\x64-windows\include\taglib\tiostream.h(53): warning C4251: 'TagLib::FileName::m_name': class 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>' needs to have dll-interface to be used by clients of class 'TagLib::FileName'
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\include\xstring(4871): note: see declaration of 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>'
Warning: C:\buildenv\mixxx-deps-rel-2.4-x64-windows-3e909e2\installed\x64-windows\include\taglib\tiostream.h(54): warning C4251: 'TagLib::FileName::m_wname': class 'std::basic_string<wchar_t,std::char_traits<wchar_t>,std::allocator<wchar_t>>' needs to have dll-interface to be used by clients of class 'TagLib::FileName'
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\include\xstring(4872): note: see declaration of 'std::basic_string<wchar_t,std::char_traits<wchar_t>,std::allocator<wchar_t>>'
Warning: C:\buildenv\mixxx-deps-rel-2.4-x64-windows-3e909e2\installed\x64-windows\include\taglib\mp4atom.h(88): warning C4251: 'TagLib::MP4::Atom::children': class 'TagLib::List<TagLib::MP4::Atom *>' needs to have dll-interface to be used by clients of class 'TagLib::MP4::Atom'
C:\buildenv\mixxx-deps-rel-2.4-x64-windows-3e909e2\installed\x64-windows\include\taglib\mp4atom.h(41): note: see declaration of 'TagLib::List<TagLib::MP4::Atom *>'
Warning: C:\buildenv\mixxx-deps-rel-2.4-x64-windows-3e909e2\installed\x64-windows\include\taglib\mp4atom.h(102): warning C4251: 'TagLib::MP4::Atoms::atoms': class 'TagLib::List<TagLib::MP4::Atom *>' needs to have dll-interface to be used by clients of class 'TagLib::MP4::Atoms'
C:\buildenv\mixxx-deps-rel-2.4-x64-windows-3e909e2\installed\x64-windows\include\taglib\mp4atom.h(41): note: see declaration of 'TagLib::List<TagLib::MP4::Atom *>'
```

The common workaround seems to disable this warning. A upstream solution would be better.

**Comment by ufleisch** [comment]
Unfortunately, there are even a lot more such warnings in the current master branch because we now use `std::unique_ptr` for the pimpls. I have fixed it by introducing `__declspec(dllexport)` for every public and protected member instead of for the whole class - probably the only way to avoid private members to be exported. There is now only a single warning suppression left in _tpropertymap.cpp_, because there seems to be no way to export an explicit template instantiation without setting `__declspec(dllexport)` on the whole class.

**Comment by ufleisch** [comment]
Fixed in version 2.0.
